File: app/daydata.py
class DayData:
  date=""
  open=None
  high=None
  low=None
  close=None
  portfolio=0
  def __init__(self, date_):
    self.date = date_
  def to_dict(self):
    # used for easy conversion to dataframe
    return{
      'date': self.date,
      'open': self.open,
      'close': self.close,
      'portfolio':self.portfolio
    }

File: app_cpp/app_cpp/daydata.cpp
#include "daydata.h"

daydata::daydata(std::chrono::year_month_day date) : date_(date){}

std::chrono::year_month_day daydata::get_date(){return date_;}

void daydata::set_open(double open){open_=open;}
double daydata::get_open(){return open_;}

void daydata::set_high(double high){high_=high;}
double daydata::get_high(){return high_;}

void daydata::set_low(double low){low_=low;}
double daydata::get_low(){return low_;}

void daydata::set_close(double close){close_=close;}
double daydata::get_close(){return close_;}

void daydata::set_portfolio(double portfolio_val){portfolio_ = portfolio_val;}
double daydata::get_portfolio(){return portfolio_;}